api/src/main/java/org/openmrs/module/ipd/api/service/impl/WardServiceImpl.java

```diff
@@ -35,7 +36,15 @@ public WardPatientsSummary getIPDWardPatientSummary(String wardUuid) {
     @Override
     public List<AdmittedPatient> getWardPatientsByUuid(String wardUuid) {
         Location location= Context.getService(LocationService.class).getLocationByUuid(wardUuid);
-        return wardDAO.getAdmittedPatientsByLocation(location);
+        return wardDAO.getAdmittedPatients(location,null, null);
+    }
+
+    @Override
+    public List<AdmittedPatient> getProviderWardPatientsByUuid(String wardUuid, String providerUuid) {
+        Location location = Context.getService(LocationService.class).getLocationByUuid(wardUuid);
+        Provider provider = Context.getProviderService().getProviderByUuid(providerUuid);
+        Date currentDateTime = new Date();
+        return wardDAO.getAdmittedPatients(location, provider, currentDateTime);
     }
```

Comment:
Also one doubt, is there any reason for creating new service method instead of refactoring the existing method ?